pkg/apis/networking/types.go

```diff
@@ -313,7 +313,39 @@ type IngressClassSpec struct {
 	// configuration for the controller. This is optional if the controller does
 	// not require extra parameters.
 	// +optional
-	Parameters *api.TypedLocalObjectReference
+	Parameters *IngressClassParametersReference
+}
+
+const (
+	// IngressClassParametersReferenceScopeNamespace represents a Namespace
+	// scoped ParametersReference resource.
```

Comment:
Nit: "Namespace scoped ParametersReference" is a bit confusing to me, maybe something like this:
```suggestion
	// IngressClassParametersReferenceScopeNamespace indicates that
	// the referenced Parameters resource is namespace-scoped.
```